src/WebJobs.Script.WebHost/Configuration/ExtensionSystemOptionsSetup.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
+{
+    public class ExtensionSystemOptionsSetup : IConfigureNamedOptions<ExtensionSystemOptions>
+    {
+        private const string SystemSectionName = "system";
+        private readonly IConfiguration _configuration;
+
+        public ExtensionSystemOptionsSetup(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void Configure(string name, ExtensionSystemOptions options)
+        {
+            if (!string.IsNullOrEmpty(name))
+            {
+                IConfigurationSection section = _configuration.GetSection(ConfigurationPath.Combine(ConfigurationSectionNames.JobHost, ConfigurationSectionNames.Extensions, name, SystemSectionName));
+                section?.Bind(options);
+                options.ExtensionName = name;
+            }
+        }
+
+        public void Configure(ExtensionSystemOptions options)
+        {
+            Configure(null, options);
```

Comment:
tiny nit: I use this instead of `null` here:


```suggestion
            Configure(Options.DefaultName, options);
```

It is just `string.Empty`.